core/evalir.ml

```diff
@@ -10,6 +10,24 @@ open Var
 let lookup_fun = Tables.lookup Tables.fun_defs
 let find_fun = Tables.find Tables.fun_defs
 
+    (* is this the best place for these? *)
+let value_as_string db =
+  function
+    | `String s -> "\'" ^ db # escape_string s ^ "\'"
+    | v -> Value.string_of_value v
+
+let row_columns : Value.t -> string list = function
+  | `List ((`Record fields)::_) -> List.map fst fields
+  | r -> failwith ("Internal error: forming query from non-row (row_columns): "^ Value.string_of_value r)
+
+let row_values db = function
+  | `List records ->
+      (List.map (function
+        | `Record fields -> List.map (value_as_string db -<- snd) fields
+        | _ -> failwith "Internal error: forming query from non-row") records)
```

Comment:
Could you use `Errors.internal_error` instead of `failwith` here?

Reply:
Thanks. The `failwith`s were what was there in the original code (this is copy-pasted from `lib.ml`).  But this seems like a reasonable time to kill them off.